engine/src/main/java/org/terasology/world/block/loader/BlockFamilyDefinitionFormat.java

```diff
@@ -70,22 +65,21 @@
 
     private static final ResourceUrn DEFAULT_SOUNDS = new ResourceUrn("engine", "default");
 
-    private final BlockFamilyFactory symmetricFamily = new SymmetricBlockFamilyFactory();
-    private final BlockFamilyFactory horizontalFamily = new HorizontalBlockFamilyFactory();
-    private final BlockFamilyFactory freeformFamily = new FreeformBlockFamilyFactory();
     private final AssetManager assetManager;
     private final Gson gson;
+    private final BlockFamilyRegistry blockFamilyRegistry;
 
-    public BlockFamilyDefinitionFormat(AssetManager assetManager, BlockFamilyFactoryRegistry blockFamilyFactoryRegistry) {
+    public BlockFamilyDefinitionFormat(AssetManager assetManager, BlockFamilyRegistry blockFamilyRegistry) {
         super("block");
         this.assetManager = assetManager;
+        this.blockFamilyRegistry = blockFamilyRegistry;
         gson = new GsonBuilder()
                 .registerTypeAdapterFactory(new CaseInsensitiveEnumTypeAdapterFactory())
                 .registerTypeAdapterFactory(new AssetTypeAdapterFactory(assetManager))
                 .registerTypeAdapter(BlockFamilyDefinitionData.class, new BlockFamilyDefinitionDataHandler())
                 .registerTypeAdapter(Vector3f.class, new Vector3fTypeAdapter())
                 .registerTypeAdapter(Vector4f.class, new Vector4fTypeAdapter())
-                .registerTypeAdapter(BlockFamilyFactory.class, new BlockFamilyFactoryHandler(blockFamilyFactoryRegistry))
+                .registerTypeAdapter(Class.class, new BlockFamilyHandler(blockFamilyRegistry))
```

Comment:
This is throwing a big red flag for me -- is there a way to limit it to subtypes of a certain class? I guess it's not so bad if it's limited to  just block family definitions